Specs/Overlay Support Spec.md

```diff
@@ -0,0 +1,106 @@
+# Feature Specification: Overlays Support
+
+## Objective
+Support use of Overlays for enabling developers to enhance existing OpenAPI description files without changing the original file.
+
+## Problem Statement
+Existing OpenAPI documents used for AI plugin creation might lack necessary properties or require modifications for them to provide a high quality AI plugin. Direct editing of the original OpenAPI document is often undesirable or impractical.
+
+## Solution Overview
+Overlays provide a flexible mechanism for extending OpenAPI documents without directly modifying the original files. This allows necessary properties to be added, existing values modified, and the OpenAPI description tailored for effective AI plugin development.
+
+## Functional Requirements
+1. **Overlay File Format Support**
+   - Support overlay files in accordance with the Overlay Specification (v1.0.0), available at [Overlay Specification](https://github.com/OAI/Overlay-Specification/blob/3f398c6/versions/1.0.0.md).
+   - Support both YAML and JSON formats for overlay files.
+
+2. **Overlay Application**
+   - Apply overlay file to an existing OpenAPI document to produce a hybrid of the overlay file and the original OpenAPI description file without changing the original file.
+   - Ensure resulting hybrid OpenAPI document remain valid according to the OpenAPI Specification.
+
+3. **Overlay Operations**
+   - Support addition, modification, and removal of properties to the target OpenAPI document.
+
+## Example: Applying Overlay to Modify OpenAPI Description File
+
+**Original OpenAPI Description File**
+
+```yaml
+openapi: 3.0.0
+info:
+  title: Sample API
+  version: 1.0.0
+paths:
+  /users:
+    get:
+      summary: Get a list of users
+      responses:
+        '200':
+          description: Successful response
+  /users/{id}:
+    get:
+      summary: Get a user by ID
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: User ID
+      responses:
+        '200':
+          description: Successful response
+```
+
+**Overlay File (overlay.yaml)**
```

Comment:
This overlay doesn't use JSON Path to define the structure. Are we saying that this is our MVP for now? Only structured overlays vs. targeted overlays would be supported?

https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md#targeted-overlays

I'm honestly OK at this point for either, but make it clear would be better and provide a more appropriate debate!

Reply:
You can't really separate the two. Targeted overlays use a merge patch at the target. Structured overlays are just the trivial case of targeted.

Reply:
@sebastienlevert  is your question whether we will support either structured or targeted rather than both? We will support both cases.